pwndbg/commands/heap.py

```diff
@@ -23,37 +32,44 @@
 from pwndbg.heap.ptmalloc import Bins
 from pwndbg.heap.ptmalloc import BinType
 from pwndbg.heap.ptmalloc import Chunk
-from pwndbg.heap.ptmalloc import DebugSymsHeap
+from pwndbg.heap.ptmalloc import GlibcMemoryAllocator
 from pwndbg.heap.ptmalloc import Heap
 
 
-def read_chunk(addr):
+def read_chunk(addr: int) -> Dict[str, int]:
     """Read a chunk's metadata."""
     # In GLIBC versions <= 2.24 the `mchunk_[prev_]size` field was named `[prev_]size`.
     # To support both versions, change the new names to the old ones here so that
     # the rest of the code can deal with uniform names.
+    assert isinstance(pwndbg.heap.current, GlibcMemoryAllocator)
+    assert pwndbg.heap.current.malloc_chunk is not None
     renames = {
         "mchunk_size": "size",
         "mchunk_prev_size": "prev_size",
     }
-    if isinstance(pwndbg.heap.current, DebugSymsHeap):
-        val = pwndbg.gdblib.typeinfo.read_gdbvalue("struct malloc_chunk", addr)
+    if isinstance(pwndbg.heap.current.malloc_chunk, gdb.Type):
+        val = pwndbg.gdblib.memory.poi(pwndbg.heap.current.malloc_chunk, addr)
     else:
         val = pwndbg.heap.current.malloc_chunk(addr)
-    return {renames.get(key, key): int(val[key]) for key in val.type.keys()}
+    value_keys: List[str] = val.type.keys()
+    return {renames.get(key, key): int(val[key]) for key in value_keys}
 
 
-def format_bin(bins: Bins, verbose=False, offset=None):
+def format_bin(bins: Bins, verbose: bool = False, offset: int | None = None) -> List[str]:
+    assert isinstance(pwndbg.heap.current, GlibcMemoryAllocator)
```

Comment:
I think this is fine for this PR, but we should really think about how we can remove all these `assert isinstance(pwndbg.heap.current, GlibcMemoryAllocator)` checks in this PR. Maybe this will happen naturally with the jemalloc GSoC project implementing a new allocator, not sure.